apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -231,9 +236,9 @@ export class RushCommandLineParser extends CommandLineParser {
         this.addAction(new BulkScriptAction({
           actionName: command.name,
 
-          // The rush rebuild and rush build command invoke the same NPM script because they share the same
-          // package-deps-hash state.
-          commandToRun: command.name === RushConstants.rebuildCommandName ? 'build' : undefined,
+          // By default, the "rebuild" action runs the "build" script. However, if the command-line.json file
+          // overrides "rebuild," the "rebuild" script should be run.
+          commandToRun: commandToRun,
```

Comment:
This is a significant behavioral change.  Before they shared the same package-deps-hash state, like this:
```
$ rush rebuild  # takes 3 minutes
$ rush build  # finishes instantly without doing any work
```

But now that's true ONLY if you didn't customize the "rush rebuild" command.  That might be a little counterintuitive.  For example, what if the customization was just changing some CLI description text?

We probably can't fix that in this PR.  Long term we've been wanting to deprecate `rush rebuild` and provide a more generalized way to clean the incremental build state, that can be applied to any command (and @D4N14L 's phases),

For this PR, maybe we just document this nuance in a command-line.json comment.